include/sparrow/buffer_adaptor.hpp

```diff
@@ -0,0 +1,499 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cmath>
+#include <limits>
+#include <utility>
+
+#include "sparrow/buffer.hpp"
+#include "sparrow/contracts.hpp"
+
+namespace sparrow
+{
+    /**
+     * @brief Class which have internally a reference to a buffer<T> and provides the API of a buffer<U>
```

Comment:
From the implementation and the requirements below, I guess you meant it holds a a reference to a `buffer<U>` and provides an API to a `buffer<T>`? what do you think of renaming the template parameters into From and To to avoid any confusion?

Reply:
Good idea !